src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -634,15 +657,15 @@ def fieldI(field):
     if I0 in field:
         return field
     field_a = field[I0]
-    field_b = field_a.absolute_field('b0')
+    field_b = field_a.absolute_field('b_0')
     b0 = field_b.gen()
     q = b0.minpoly()
     qembd = field_b.embeddings(QQbar)
     for h1 in qembd:
         b1 = h1(b0)
         b2 = h1(field_b(field_a.gen(0)))
-        b3 = field.gen(0)
-        F1 = NumberField(q, 'b', embedding=b1)
+        b3 = QQbar(field.gen(0))
```

Comment:
Why this change? This could make things more costly to compute. I imagine you have a good reason, and if so, you should add a doctest showing what didn't work before (or wouldn't work without this change).

Reply:
During the changes I tested with some other fields, cyclotomic, quadratic. These fields are subfields of QQbar but without imposiing b3 is in QQbar it did not work.

Reply:
Then please add the appropriate tests (if you have already, you can just state it; I haven't looked that closely yet).

Reply:
Test added but I do not know how to make the addtion of `QQbar` relevant.

Reply:
This is good. The test was not working before. Although the location of the test is in the wrong position since it is the block showing if `I` is in the field.

I might argue that it would be reasonable to treat cyclotomic fields as special since the corresponding extension would again be a cyclotomic field by scaling the order to be a multiple of 4. However, that might be orthogonal to the issue at hand.

Reply:
I changed the position of the test. I saw that this addition was also relevant to quadratic fields but finally I chose only cyclotomic for the test.

Reply:
Can you add one of those?

Although this has a slight code smell with possibly hiding a bug.